libdrizzle/pack.cc

```diff
@@ -481,3 +481,24 @@ static drizzle_return_t _pack_scramble_hash(drizzle_st *con,
 
   return DRIZZLE_RETURN_OK;
 }
+
+void drizzle_unpack_error(drizzle_st *con)
```

Comment:
Yes. But might have to change the logic a bit while the currently `drizzle_unpack_error` sets the `packet_size = 0`, which is not done in the error parsing in `result.cc`